dist.ts

```diff
@@ -0,0 +1,196 @@
+import { Temporal } from "@js-temporal/polyfill";
+import {
+  BASELINE_LOW_TO_HIGH_DURATION,
+  computeBaseline,
+} from "compute-baseline";
+import { Compat, Feature } from "compute-baseline/browser-compat-data";
+import assert from "node:assert/strict";
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { isDeepStrictEqual } from "node:util";
+import YAML, { Document } from "yaml";
+
+/**
+ * Update (or create) a dist YAML file from a feature definition YAML file.
+ *
+ * @param {string} fp The path to the human-authored YAML file.
+ */
+function updateDistFile(fp: string): void {
+  const { dir, name: id } = path.parse(fp);
+  const distPath = path.join(dir, `${id}.dist.yml`);
+
+  const dist = toDist(fp);
+  fs.writeFileSync(distPath, dist.toString({ lineWidth: 0 }));
+}
+
+/**
+ * Generate a dist YAML document from a feature definition YAML file path.
+ *
+ * This passes through most of the contents of a feature definition. If
+ * possible, it fills in `compat_features` from @mdn/browser-compat-data and, if
+ * successful, generates a `status` block.
+ */
+function toDist(fp: string): YAML.Document {
+  const yaml = YAML.parseDocument(fs.readFileSync(fp, { encoding: "utf-8" }));
+  const { name: id } = path.parse(fp);
+
+  const taggedCompatFeatures = (
+    tagsToFeatures.get(`web-features:${id}`) ?? []
+  ).map((f) => `${f.id}`);
+
+  const overridden = {
+    compatFeatures: yaml.toJS().compat_features,
+    status: resolveBaselineHighDate(yaml.toJS().status),
+  };
+
+  const generated = {
+    compatFeatures: taggedCompatFeatures.length
+      ? taggedCompatFeatures
+      : undefined,
+    status: taggedCompatFeatures.length
+      ? JSON.parse(
+          computeBaseline({
+            compatKeys: taggedCompatFeatures as [string, ...string[]],
+            checkAncestors: false,
+          }).toJSON(),
+        )
+      : undefined,
+    statusByCompatFeaturesOverride: Array.isArray(overridden.compatFeatures)
+      ? JSON.parse(
+          computeBaseline({
+            compatKeys: overridden.compatFeatures as [string, ...string[]],
+            checkAncestors: false,
+          }).toJSON(),
+        )
+      : undefined,
+  };
+
+  warnOnNeedlessOverrides(id, overridden, generated);
+
+  if (overridden.status?.baseline === "high") {
+    insertBaselineHighDate(yaml, overridden.status.baseline_high_date);
+  }
+
+  if (!overridden.compatFeatures && generated.compatFeatures) {
+    insertCompatFeatures(yaml, generated.compatFeatures);
+  }
+
+  if (!overridden.status) {
+    const status = generated.statusByCompatFeaturesOverride ?? generated.status;
+    if (status) {
+      insertStatus(yaml, status);
+    }
+  }
+
+  insertHeaderComments(yaml, id);
+
+  return yaml;
+}
```

Comment:
@foolip in trying to emit a warning when status overrides are needless, I "accidentally" refactored this to separate the "find the highest source of truth" bit from the "edit YAML" bit.

This doesn't change the results of the writing dists (at least for the set of files that I've generated dists for—a nice benefit of this approach is that Git now tells me whether I've modified those files). But the code is quite different. If you want me to back this out and submit it as a separate PR rather than re-review, let me know.

Reply:
I've reviewed it, looks good!